mono/mini/intrinsics.c

```diff
@@ -912,6 +912,15 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			EMIT_NEW_UNALU (cfg, ins, OP_ICGT, dreg, -1);
 			ins->type = STACK_I4;
 			return ins;
+		} else if (!strcmp (cmethod->name, "ArrayObjectHasReferences")) {
```

Comment:
Rename to `HasReferences` here?

Reply:
I think it makes sense only for Arrays - other objects will be boxed and always return true

Reply:
I assumed it would work for boxed value types but okay.

Reply:
I could make it a generic "HasReferences(object)" API, but the problem is - Array itself is a GC type so the API would be confusing - e.g. `HasReference(new int[] {1})` = false.

Reply:
Let me know if you need a generic API (where any Array will be `true` but custom structs without GC types will be `false`)